### DISCUSSION TITLE
Question about setting the values for CMAKE_CUDA_ARCHITECTURES

### DISCUSSION THREAD

**Discussion by Qubitium** [NVIDIA/nvbench#129] [Q&A]
What are the values for ```CMAKE_CUDA_ARCHITECTURES```? For example, for Ampere 3090 series. 

Cannot find this value anywhere. Thanks.

**Comment by jrhemstad** [comment]
`CMAKE_CUDA_ARCHITECTURES` is a CMake option documented here: https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_ARCHITECTURES.html

It is used as the default value for [`CUDA_ARCHITECTURES`](https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html#prop_tgt:CUDA_ARCHITECTURES) on a CUDA target. 

This option is akin to the architecture specification options in nvcc: https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#base-notation

You can look up the compute capability of a GPU here: https://developer.nvidia.com/cuda-gpus

So for your example of a 3090, the compute capability is 8.6. So to build both PTX and SASS for that GPU you would specify `-DCMAKE_CUDA_ARCHITECTURES=86`. 

A nice feature starting with CMake 3.24 is the ability to say `-DCMAKE_CUDA_ARCHITECTURES=native` which will automatically detect the GPU on your system and build for that GPU.

**Reply by Qubitium** [reply]
Thanks!

 I think using "75" to mean "7.5" for users is really bad. The CUDA_ARCHITECTURES use 75 to reflect "7.5" in official Nvidia documentation. The value format difference is what caused my problem of lookup what cuda 75 is since there is no cuda 75 but cuda compute capability 7.5.

75 = 7.5 is really not intuitive unless you are talking about strings. But everything here reflects a number.

**Reply by jrhemstad** [reply]
Note that this convention isn't unique to cmake. This is how CUDA and nvcc itself defines them: https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#gpu-feature-list

![image](https://user-images.githubusercontent.com/15221289/224093698-277656f0-b60b-46f7-a36a-8f58fd2581eb.png)

**Reply by Qubitium** [reply]
The table you posted causes another issue and a new format label sm_xx. So there are 3 places each referring to cuda capability use 3 different notations. I know this is semantics and nit picky but there is no reason to use 3 different formats to reflect the same value.  

* cuda architecture 75
* cuda compute capabilities 7.5
* sm_75

Stick with one and use it for everything (doc + config + codes) ?

**Reply by jrhemstad** [reply]
Perhaps, but none of this is specific to nvbench 🙂

**Reply by Qubitium** [reply]
I understand. But please kick the ball up the chain, if possible. There should be one method of naming cuda hardware revision in technical docs/code. Marketing can call it whatever they want. Another thing for tech/code.
